### ISSUE TITLE
Deleted & renamed pages still being reccomended in search results

### ISSUE THREAD

**Issue by engageintellect** [shuding/nextra#1279]
Not sure if this is an issue, or if I am missing something, but...

I create a page called "helloworld.mdx". After a few minutes, search has indexed it and it shows up in the search results. Now, if I rename or delete this page it will continue to show up in the search results.

How can I make it so only existing/active pages are show in search results?

TIA

**Comment by rbox-risk** [comment]
I've noticed this while developing. In chrome, disabling network cache and reloading appears to fix the issue. As to what the root cause is, maybe asset cache/expiry header settings?

**Comment by engageintellect** [comment]
> I've noticed this while developing. In chrome, disabling network cache and reloading appears to fix the issue. As to what the root cause is, maybe asset cache/expiry header settings?

Hmmm, I just deployed my project to Vercel, and visited the link from an incognito browser instance... cached search results are still there. Super annoying. 😩

**Comment by shuding** [comment]
I'm planning a rewrite of the search indexing this week 🙏

As a workaround you can redeploy with cache disabled.

**Comment by rbox-risk** [comment]
Awesome to hear that. Given that,I thought I'd mention this. If I add specific keywords in an `.mdx` using a hidden `div` directly, then search will find these


```
export const kwords = ["componentWord", "componentAwesome"];

<KeywordPages hide={false} keywords={kwords} />

Markdown stuff

<div className="hidden">
  <p>❓ pageSpecificWord</p>
  <p>❓ awesomeKeyword</p>
</div>
```

![image](https://user-images.githubusercontent.com/59801351/213039330-a580d9b0-fa29-47e6-8af5-8c525d6e568a.png)

But If wrap the div in a component, unroll the array of keywords in the usual way, search will not pick up the keyword.

![image](https://user-images.githubusercontent.com/59801351/213039806-13ccce90-debe-419c-8791-918b6ebd7603.png)

I guess this is simply a result of how/when the tree traversal needs to happen in producing the search index at build time?

The other reason for mentioning is, perhaps there's a more idiomatic way to add keywords to pages that we don't want to be visible but want a user searching for said terms to find a specific page (frontmatter, or `_meta` perhaps)?

**Comment by shuding** [comment]
> I guess this is simply a result of how/when the tree traversal needs to happen in producing the search index at build time?

Yes exactly! It can only index things in the AST, but not dynamic things inside functions (a component is a function).

> The other reason for mentioning is, perhaps there's a more idiomatic way to add keywords to pages that we don't want to be visible but want a user searching for said terms to find a specific page (frontmatter, or _meta perhaps)?

This can be a great separate issue, would you mind writing an example in a new issue? Thanks!

**Comment by rbox-risk** [comment]
> > I guess this is simply a result of how/when the tree traversal needs to happen in producing the search index at build time?
> 
> Yes exactly! It can only index things in the AST, but not dynamic things inside functions (a component is a function).
> 
> > The other reason for mentioning is, perhaps there's a more idiomatic way to add keywords to pages that we don't want to be visible but want a user searching for said terms to find a specific page (frontmatter, or _meta perhaps)?
> 
> This can be a great separate issue, would you mind writing an example in a new issue? Thanks!

https://github.com/shuding/nextra/issues/1299#issue-1548085050
thanks

**Comment by engageintellect** [comment]
> Closing this one with #1290 as problems related to search index caching should be gone!

I appreciate the promptness. This fix really helped me out. Thank you 🙏
